qemu/tests/smt_test.py

```diff
@@ -0,0 +1,120 @@
+        if session.cmd_status(test_installed_cmd) != 0:
+            logging.warning("Could not find installed heavyload in guest, will"
+                            " install it via winutils.iso ")
+            winutil_drive = utils_misc.get_winutils_vol(session)
+            if not winutil_drive:
+                test.cancel("WIN_UTILS CDROM not found.")
+            install_cmd = params["install_cmd"] % winutil_drive
+            session.cmd(install_cmd)
+
+    def run_stress():
+        """
+        Run stress inside guest, return guest cpu usage
+        """
+        error_context.context("Run stress in guest and get cpu usage",
+                              logging.info)
+        if os_type == "linux":
+            stress_args = params["stress_args"]
+            stress_test = utils_test.VMStress(vm, "stress",
+                                              params, stress_args=stress_args)
+            try:
+                stress_test.load_stress_tool()
+                time.sleep(stress_duration / 2)
+                output = session.cmd_output_safe(params["get_cpu_usage_cmd"])
+                utils_misc.wait_for(lambda: (stress_test.app_running is False), 30)
+                stress_test.unload_stress()
+                cpu_usage = re.findall(r":\s*(\d+.?\d+)\s*us", output)
+                cpu_usage = [float(x) for x in cpu_usage]
+                logging.info("Guest cpu usage is %s", cpu_usage)
+                unloaded_cpu = [x for x in cpu_usage if x < 20]
+                if unloaded_cpu:
+                    test.fail("CPU(s) load percentage is less than 20%")
+            finally:
+                stress_test.clean()
+        else:
+            install_path = params["install_path"]
+            heavyload_install(install_path)
+            error_context.context("Run heavyload inside guest.", logging.info)
+            heavyload_bin = r'"%s\heavyload.exe" ' % install_path
+            heavyload_options = ["/CPU %d" % vm.cpuinfo.smp,
+                                 "/DURATION %d" % (stress_duration // 60),
+                                 "/AUTOEXIT",
+                                 "/START"]
+            start_cmd = heavyload_bin + " ".join(heavyload_options)
+            stress_tool = BackgroundTest(session.cmd, (start_cmd,
+                                                       stress_duration,
+                                                       stress_duration))
+            stress_tool.start()
+            if not utils_misc.wait_for(stress_tool.is_alive, stress_duration):
+                test.error("Failed to start heavyload process.")
+            stress_tool.join(stress_duration)
+
+    check_cpu_flags(params, "topoext", test)
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+
+    os_type = params["os_type"]
+    stress_duration = params.get_numeric("stress_duration", 60)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
```

Comment:
Not necessary when you have `wait_for_login()` followed up.

Reply:
Thanks, updated.